rates_api/app.py

```diff
@@ -12,47 +12,61 @@
 
 @app.route("/rates")
 def get_rates() -> Response:
+    """
+    Function extracts arguments from the query parameters, checks them,
+    requests data from database and forms json object
+    :param date_from: first date of the requested period
+    :param date_to: last date of the requested period
+    :param org: origin port code
+    :param dst: destination port code
+    :return: Json object included days and average prices or error info
+    """
     date_from = request.args.get("date_from", "", type=str)
     date_to = request.args.get("date_to", "", type=str)
     orig = request.args.get("origin", "", type=str)
     dest = request.args.get("destination", "", type=str)
 
+    # If at least one of dates is incorrect or date from > date_to:
     if not dates_validator(date_from=date_from, date_to=date_to):
-        abort(404)
-
-    available_date_points = get_available_dates(date_from, date_to, orig, dest)
-    if not available_date_points:
-        abort(404)
+        abort(422)
     else:
-        date_from_f, date_to_f = available_date_points
-        daily_rates = get_daily_rates(orig, dest)
-        if not daily_rates:
-            abort(404)
-        dates_range = get_dates(str(date_from_f), str(date_to_f))
+        # List of requested dates in string format:
+        dates_range = get_dates(str(date_from), str(date_to))
```

Comment:
should we try to use the `CTE` approach to just iterate over the DB entries and prepare them for output? it could simplify this endpoint a lot